extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsLoadSpec.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Druid - a distributed column store.
+ * Copyright 2012 - 2015 Metamarkets Group Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.storage.hdfs;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.google.common.base.Preconditions;
+import com.metamx.common.ISE;
+import io.druid.segment.loading.DataSegmentPuller;
+import io.druid.segment.loading.LoadSpec;
+import io.druid.segment.loading.SegmentLoadingException;
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.util.Map;
+
+/**
+ *
+ */
+@JsonTypeName(HdfsStorageDruidModule.SCHEME)
+public class HdfsLoadSpec implements LoadSpec
+{
+  private final Path path;
+  final HdfsDataSegmentPuller puller;
+  @JsonCreator
+  public HdfsLoadSpec(
+      @JacksonInject Map<String, DataSegmentPuller> pullers,
```

Comment:
I changed this to inject the puller directly instead of the puller map. I went against making it `@Nullable` because the LoadSpec remains a `Map<String, Object>` until it is immediately ready for consumption. Anywhere that actually consumes a LoadSpec also has to have the module loaded that the LoadSpec is a part of. This means that anything that will actually consume a HdfsLoadSpec must have the HDFS storage module added, and the same for the S3 and Cassandra modules, so I'm proposing  the rule:

Don't materialize a Map<String, Object> into a LoadSpec unless you're ready to consume it.

Which alleviates dependency issues because anyone who is ready to consume it sure better have the correct module loaded.